patcher.sh

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env bash
+
+# exit on any error to avoid showing everything was successfull even tho it wasnt
+set -eou pipefail
+
+# Patching user variables
+OSCHOICE=""
+AM2RZIP=""
+HQMUSIC=false
+SYSTEMWIDE=false
+APPIMAGE=false
+PREFIX=""
+
+# Patching internal variables
+# Since people are likely to double click on this, I need a way to get the script_dir
+dirResult="$(dirname '${BASH_SOURCE[0]}')"
+SCRIPT_DIR="$(realpath $dirResult)"
+VERSION="1.5.5"
+OUTPUT="${SCRIPT_DIR}/am2r_${VERSION}"
+RESOURCES="${OUTPUT}/assets"
+
+patch_am2r ()
+{
+	# Set prefix to default value if empty
+	if [ -z "$PREFIX" ]; then
+		if [ "$SYSTEMWIDE" = true ]; then
+			PREFIX="/usr/local"
+		else
+			PREFIX="$SCRIPT_DIR"
+		fi
+	fi
+
+	# We need to do variable adjustments based on the prefix
+	if [ "$SYSTEMWIDE" = true ]; then
+		OUTPUT="${PREFIX}/opt/am2r"
+		RESOURCES="${OUTPUT}/assets"
+	else
+		OUTPUT="${PREFIX}/am2r_${VERSION}"
+		RESOURCES="${OUTPUT}/assets"
+	fi
+	# Cleanup in case the dirs/files exists
+	cleanupDirs=("$OUTPUT" "$SCRIPT_DIR/AM2R.AppDir" "$SCRIPT_DIR/AM2R-x86_64.AppImage" "$SCRIPT_DIR/AM2RWrapper/" "$SCRIPT_DIR/utilities/android/assets/" "$SCRIPT_DIR/AM2RWrapper.apk")
+	rm -rf "${cleanupDirs[@]}"
+
+	mkdir -p "$OUTPUT"
+	mkdir -p "$RESOURCES"
+
+	# Check for AM2R_11
+	if [[ -f "$AM2RZIP" ]]; then
+		echo "AM2R_11.zip found!"
+		unzip -q "$AM2RZIP" -d "$OUTPUT"
```

Comment:
> seeing how the final output the script produces doesn't get pushed to it.

Hm, from what I can see, it does partially.
For linux local, the resulting stuff will be in $OUTPUT (i.e. OUTPUT/runner, OUTPUT/assets, OUTPUT/.desktop etc.)
For Linux systemwide, the resulting stuff will be in $OUTPUT (i.e. $OUTPUT/am2r.appimage, OUTPUT/icon.png etc.)

Android as far as i can see is the only path that places its stuff into SCRIPT_DIR instead of OUTPUT.

Reply:
For consistency I think it'd be best to use the `PREFIX` variable in those cases as well. `PREFIX` should always contain the path to the final output this script produces, I think that would make the script a lot more readable.

Reply:
Since the contents of the zip are always copied to create a clean mutable temporary directory, i do think `mktemp -d` makes sense here.

Reply:
Hm, not that sure about that. Currently OUTPUT exists inside PREFIX.
I.e., for a systemwide installation, you have something akin to this:
```
/
|-bin/
  |-...
|-opt/
  |-...
|-share/
  |-...
```
Where `/` is $PREFIX, and `/opt` is $OUTPUT
For a local installation, it'd be more like
```
~/patcher/
|-am2r_15-5
   |-...
```
where `~/patcher/` is $PREFIX and `am2r_15-5` is $OUTPUT.

Maybe it'd make sense to rename OUTPUT to something else? Since PREFIX is the place where all relevant game files with reside, with OUTPUT actually being more the place for the game data files. Something like `$GAMEDIR` perhaps?

Reply:
`GAMEDIR` sounds good yeah, or `ASSETS` maybe?

Reply:
I'll rename it to GAMEDIR, as ASSETS is kinda misleading with the RESOURCES variable (which does point to "$output/assets)"